base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -673,6 +694,143 @@ where
         Ok(tx_id)
     }
 
+    /// Sends a one side payment transaction to a recipient
+    /// # Arguments
+    /// 'dest_pubkey': The Comms pubkey of the recipient node
+    /// 'amount': The amount of Tari to send to the recipient
+    /// 'fee_per_gram': The amount of fee per transaction gram to be included in transaction
+    pub async fn send_one_sided_transaction(
+        &mut self,
+        dest_pubkey: CommsPublicKey,
+        amount: MicroTari,
+        fee_per_gram: MicroTari,
+        message: String,
+        transaction_broadcast_join_handles: &mut FuturesUnordered<
+            JoinHandle<Result<u64, TransactionServiceProtocolError>>,
+        >,
+    ) -> Result<TxId, TransactionServiceError>
+    {
+        if self.node_identity.public_key() == &dest_pubkey {
+            warn!(target: LOG_TARGET, "One-sided spend-to-self transactions not supported");
+            return Err(TransactionServiceError::OneSidedTransactionError(
+                "One-sided spend-to-self transactions not supported".to_string(),
+            ));
+        }
+
+        // Prepare sender part of the transaction
+
+        let mut stp = self
+            .output_manager_service
+            .prepare_transaction_to_send(
+                amount,
+                fee_per_gram,
+                None,
+                message.clone(),
+                script!(PushPubKey(Box::new(dest_pubkey.clone()))),
+            )
+            .await?;
+        let tx_id = stp.get_tx_id()?;
+        if !stp.is_single_round_message_ready() {
+            return Err(TransactionServiceError::OneSidedTransactionError(
+                "Sender protocol single round message not ready".to_string(),
+            ));
+        }
+
+        let _ = stp
+            .build_single_round_message()
+            .map_err(|e| TransactionServiceProtocolError::new(tx_id, TransactionServiceError::from(e)))?;
+        if !stp.is_collecting_single_signature() {
```

Comment:
Under what conditions is this check not met?

Reply:
Please see previous comment